File: library/src/main/java/com/yzyfdf/library/utils/ImageUtil.java
package com.yzyfdf.library.utils;

import android.content.Context;
import android.content.Intent;
import android.net.Uri;

import java.io.File;

/**
 * @author sjj , 2019/5/23 16:55
 */
public class ImageUtil {
    private static volatile ImageUtil sInstance;

    private ImageUtil() {
    }

    public static ImageUtil getInstance() {

        if (sInstance == null) {
            synchronized (ImageUtil.class) {
                if (sInstance == null) {
                    sInstance = new ImageUtil();
                }
            }
        }
        return sInstance;
    }

    //更新图库
    public void updatePhotoMedia(File file, Context context) {
        Intent intent = new Intent();
        intent.setAction(Intent.ACTION_MEDIA_SCANNER_SCAN_FILE);
        intent.setData(Uri.fromFile(file));
        context.sendBroadcast(intent);
    }
}

File: library/src/main/java/com/yzyfdf/library/utils/ImageUtil.kt
package com.yzyfdf.library.utils

import android.content.Context
import android.content.Intent
import android.media.MediaScannerConnection.scanFile
import android.net.Uri
import java.io.File

/**
 * @author sjj , 2019/5/23 16:55
 */
class ImageUtil private constructor() {

    //更新图库
    fun updatePhotoMedia1(file: File, context: Context) {
        val intent = Intent()
        intent.action = Intent.ACTION_MEDIA_SCANNER_SCAN_FILE
        intent.data = Uri.fromFile(file)
        context.sendBroadcast(intent)
    }

    fun updatePhotoMedia2(file: File, context: Context) {
        scanFile(context, arrayOf(file.absolutePath), arrayOf("image/jpeg")) { _, _ -> }
    }

    companion object {
        val instant: ImageUtil by lazy {
            ImageUtil()
        }
    }
}
